src/track/trackrecord.cpp

```diff
@@ -47,4 +47,85 @@ bool TrackRecord::updateGlobalKeyText(
+            mergedTrackInfo.setTrackNumber(importedTrackInfo.getTrackNumber());
+        }
+    }
+#if defined(__EXTRA_METADATA__)
+    mergeNullableMetadataProperty(mergedTrackInfo.refConductor(), importedTrackInfo.getConductor());
+    mergeNullableMetadataProperty(mergedTrackInfo.refDiscNumber(), importedTrackInfo.getDiscNumber());
+    mergeNullableMetadataProperty(mergedTrackInfo.refDiscTotal(), importedTrackInfo.getDiscTotal());
+    mergeNullableMetadataProperty(mergedTrackInfo.refEncoder(), importedTrackInfo.getEncoder());
+    mergeNullableMetadataProperty(mergedTrackInfo.refEncoderSettings(), importedTrackInfo.getEncoderSettings());
+    mergeNullableMetadataProperty(mergedTrackInfo.refISRC(), importedTrackInfo.getISRC());
+    mergeNullableMetadataProperty(mergedTrackInfo.refLanguage(), importedTrackInfo.getLanguage());
+    mergeNullableMetadataProperty(mergedTrackInfo.refLyricist(), importedTrackInfo.getLyricist());
+    mergeNullableMetadataProperty(mergedTrackInfo.refMood(), importedTrackInfo.getMood());
+    mergeNullableMetadataProperty(mergedTrackInfo.refMovement(), importedTrackInfo.getMovement());
+    mergeNullableMetadataProperty(mergedTrackInfo.refMusicBrainzArtistId(), importedTrackInfo.getMusicBrainzArtistId());
+    mergeNullableMetadataProperty(mergedTrackInfo.refMusicBrainzRecordingId(), importedTrackInfo.getMusicBrainzRecordingId());
+    mergeNullableMetadataProperty(mergedTrackInfo.refMusicBrainzReleaseId(), importedTrackInfo.getMusicBrainzReleaseId());
+    mergeNullableMetadataProperty(mergedTrackInfo.refMusicBrainzWorkId(), importedTrackInfo.getMusicBrainzWorkId());
+    mergeNullableMetadataProperty(mergedTrackInfo.refRemixer(), importedTrackInfo.getRemixer());
+    mergeNullableMetadataProperty(mergedTrackInfo.refSubtitle(), importedTrackInfo.getSubtitle());
+    mergeNullableMetadataProperty(mergedTrackInfo.refWork(), importedTrackInfo.getWork());
+#endif // __EXTRA_METADATA__
+    AlbumInfo& mergedAlbumInfo = refMetadata().refAlbumInfo();
+    const AlbumInfo& importedAlbumInfo = importedFromFile.getAlbumInfo();
+#if defined(__EXTRA_METADATA__)
+    mergeReplayGainMetadataProperty(mergedAlbumInfo.refReplayGain(), importedAlbumInfo.getReplayGain());
+    mergeNullableMetadataProperty(mergedAlbumInfo.refCopyright(), importedAlbumInfo.getCopyright());
+    mergeNullableMetadataProperty(mergedAlbumInfo.refLicense(), importedAlbumInfo.getLicense());
+    mergeNullableMetadataProperty(mergedAlbumInfo.refMusicBrainzArtistId(), importedAlbumInfo.getMusicBrainzArtistId());
+    mergeNullableMetadataProperty(mergedAlbumInfo.refMusicBrainzReleaseGroupId(), importedAlbumInfo.getMusicBrainzReleaseGroupId());
+    mergeNullableMetadataProperty(mergedAlbumInfo.refMusicBrainzReleaseId(), importedAlbumInfo.getMusicBrainzReleaseId());
+    mergeNullableMetadataProperty(mergedAlbumInfo.refRecordLabel(), importedAlbumInfo.getRecordLabel());
+#else
+    Q_UNUSED(mergedAlbumInfo);
```

Comment:
Why can't we put the declaration also into the #if defined(__EXTRA_METADATA__) block?

Reply:
Of course. Maybe I have left it outside of the #ifdef to show that only the merge = conditional-copy operations for individual properties differ. In this (special) case the set of currently supported properties is empty.

Reply:
if there is a reason, keep it like it is.

Reply:
Not really, already simplified.